common/src/main/java/io/druid/concurrent/Execs.java

```diff
@@ -35,24 +37,52 @@
  */
 public class Execs
 {
-  public static ExecutorService singleThreaded(String nameFormat)
+  private static int nullToNormal(@Nullable Integer priority)
```

Comment:
And the code would actually be more complex since you couldn't have the cases with no priority call the cases with priority with NULL values